tools/run.py

```diff
@@ -0,0 +1,392 @@
+
+        # distribute each event in the timeline
+        time_idx = 0
+        for nl in range(nevents_in_layer):
+
+            event_duration_nl, riridx, sample_onoffsets, ev_traj, is_moving, is_flipped_moving, ev_speed, event_nl, time_idx = get_event_riridx_onoffset(nl, silence_gaps, event_idx_in_layer, fold_event_durs, fold_event_filenames, cfg, time_idx, n_traj, traj_doas)
+
+            nth_mixture = update_nth_mixture(nth_mixture, riridx, time_idx, event_duration_nl, sample_onoffsets, traj_doas, nl, layer, ev_traj, fold_event_filenames, is_moving, is_flipped_moving, ev_speed, event_nl, cfg)
+
+            time_idx += event_duration_nl
+
+    sort_idx = np.argsort(nth_mixture['event_onoffsets'][:,0])
+    nth_mixture['files'] = nth_mixture['files'][sort_idx]
+    nth_mixture['class'] = nth_mixture['class'][sort_idx]
+    nth_mixture['event_onoffsets'] = nth_mixture['event_onoffsets'][sort_idx]
+    #nth_mixture['sample_onoffsets'] = nth_mixture['sample_onoffsets'][sort_idx]
+    nth_mixture['trajectory'] = nth_mixture['trajectory'][sort_idx]
+    nth_mixture['isMoving'] = nth_mixture['isMoving'][sort_idx]
+    nth_mixture['isFlippedMoving'] = nth_mixture['isFlippedMoving'][sort_idx]
+    nth_mixture['speed'] = nth_mixture['speed'][sort_idx]
+    nth_mixture['rirs'] = np.array(nth_mixture['rirs'],dtype=object)
+    nth_mixture['rirs'] = nth_mixture['rirs'][sort_idx]
+    new_doas = np.zeros(len(sort_idx),dtype=object)
+    new_sample_onoffsets = np.zeros(len(sort_idx),dtype=object)
+    upd_idx = 0
+    for idx in sort_idx:
+        new_doas[upd_idx] = nth_mixture['doa_azel'][idx].T
+        new_sample_onoffsets[upd_idx] = nth_mixture['sample_onoffsets'][idx]
+        upd_idx += 1
+    nth_mixture['doa_azel'] = new_doas
+    nth_mixture['sample_onoffsets'] = new_sample_onoffsets 
+
+    return nth_mixture
+
+def main():
+    '''
+    main function to trigger the data generation
+    '''
+
+    # parse config arguments
+    args = parse_args()
+    cfg = load_config(args, args.path_to_config)
+    print('generating data using parameters:')
+    print(yaml.dump(dict(cfg), allow_unicode=True, default_flow_style=False))
+
+    
+    event_filenames = get_sound_event_filenames(cfg.PATH_TO_SOUND_EVENT_FILES)
+    event_durs = get_sound_event_filedurs(cfg.PATH_TO_SOUND_EVENT_DURS)
+
+
+    fold_names = cfg.FOLD_NAMES
+    snr_range = cfg.SNR_RANGE
+    max_polyphony = cfg.MAX_POLYPHONY
+    mixtures = []
+
+    # iterate over fold names
+    for ifold, fold in enumerate(fold_names):
+        
+        fold_event_filenames, fold_event_durs = get_fold_files_and_durs(
+                fold, 
+                event_filenames, 
+                event_durs)
+        n_mixtures_per_fold = cfg.N_MIX_PER_FOLD[ifold]
+
+        # iterate over rooms
+        fold_rooms = cfg.FOLD_ROOMS[fold]
+        room_mixtures=[]
+        for room_name in fold_rooms:
+            fold_mixture = {'mixture': []}
```

Comment:
I find these variable names confusing, but I'm sticking to the original ones for now. I guess `room_mixtures` is fine, but `fold_mixture` is initialized inside the room, while we are iterating over the fold in the loop around all of this.